bots/mainbot/prophet.js

```diff
@@ -7,33 +7,97 @@ const prophet = {};
-    //             self.log("Attacking Castle");
-    //             return combat.attackBot(self, enemies[i]);
-    //         }
-    //     }
-    // }
-    // if (self.step === 0 ){
-    //     self.log("Searching for castle on horizontal symmetric map");
-    //     self.moveQueue = movement.moveTo(self, (self.map[0].length - self.me.x), (self.me.y))
-    // }
-    // if (self.moveQueue.length !== 0) {
-    //     let move = self.moveQueue.shift();
-    //     self.log("moving to " + (move.x) + ', ' + (move.y));
-    //     return self.move((move.x - self.me.x), (move.y - self.me.y));
-    // }
-    // else if(self.step !== 0 && enemies.length < 1){
-    //     self.log("Searching for castle on vertically symmetric map");
-    //     self.moveQueue = movement.moveTo(self, (self.map[0].length - self.me.x), (self.map.length - self.me.y))
-    // }
+    self.castleTalk(SPECS.PROPHET)
+    if (self.horizontalCastle === undefined){
+        self.horizontalCastle = {x: (self.map[0].length - self.me.x), y: self.me.y};
+    }
+    if (self.verticalCastle === undefined){
+        self.verticalCastle = {x: self.me.x, y: (self.map.length - self.me.y)};
+    }
+    if (self.targetCastle === undefined){
+        self.targetCastle = self.horizontalCastle;
+    }
+    if(self.attacking === undefined){
+        self.attacking = false;
+    }
+    let enemies = utilities.enemiesInRange(self);
+    // If attack skip phase check
+    if (self.attacking === true){
+        utilities.log(self, "Continuing attack phase")
+        return prophet.attackPhase(self, enemies);
+    }
+    // If we aren't attacking check the phase to see if castle has said to attack
+	let phase = utilities.getCastleSignal(self);
 
+	if(phase === 1){
+        self.attacking = true;
+		return prophet.attackPhase(self, enemies);
+	}
+	else{
+		utilities.log(self, "defaulting to build phase");
+        return prophet.buildPhase(self, enemies);
+	}
+};
+
+prophet.attackPhase = (self, enemies) => {
+    utilities.log(self, "Executing Attack Phase turn");
+    if (self.moveQueue === undefined) {
+        self.moveQueue = [];
+    }
+    utilities.log(self, `Found ${enemies.length} enemies near me`)
+    if(enemies.length > 0){
+        for(let i = 0; i < enemies.length; ++i){
+            if(enemies[i].unit === SPECS.CASTLE){
+                self.log("Attacking Castle");
+            }
+            return combat.attackBot(self, enemies[i]);
+        }
+    }
+
+    // If we made it to the horizontal location and there is nothing to attack the castle is probably at the vertical loc
+    if(self.me.x === self.horizontalCastle.x && self.me.y === self.horizontalCastle.y){
+    	self.targetCastle = self.verticalCastle
+	}
+
+    if (self.moveQueue.length !== 0) {
+        let move = self.moveQueue.shift();
+        self.log("moving to " + (move.x) + ', ' + (move.y));
+        try{
+            return self.move((move.x - self.me.x), (move.y - self.me.y));
+		}
```

Comment:
Nitpick: More wacky indeinting

Reply:
I think this should be fixed in the castle-rework branch. I can pr it once we get this through. Let's handle formatting issues on that pr. Sound good?